### PR TITLE
add WriteBufferManager support

### PR DESCRIPTION
**Pull request by benoitmeriaux**
add WriteBufferManager support, which was recently added in rocksdb C API with https://github.com/facebook/rocksdb/pull/11710 and  https://github.com/facebook/rocksdb/pull/12007

### CODE REVIEW

**Comment by benoitmeriaux** [comment]
@aleksuss could you accept the workflow so I know if everything is ok and take a look, please ?

File: src/db_options.rs [hunk 1]
```diff
@@ -35,6 +35,106 @@ use crate::{
...
+    ///             It will wait for flush to complete and memory usage to drop down
+    pub fn new_write_buffer_manager(buffer_size: size_t, allow_stall: bool) -> WriteBufferManager {
+        let inner = NonNull::new(unsafe {
+            ffi::rocksdb_write_buffer_manager_create(buffer_size, allow_stall)
+        })
+        .unwrap();
+        WriteBufferManager(Arc::new(WriteBufferManagerWrapper { inner }))
+    }
+
+    /// users can set up RocksDB to cost memory used by memtables to block cache.
```
Comment (aleksuss) [inline src/db_options.rs:84]: users -> Users?

**Comment by benoitmeriaux** [comment]
gentle bump on this @aleksuss to run the workflow :)

**Comment by zaidoon1** [comment]
> gentle bump on this @aleksuss to run the workflow :)
There is still some clippy warning, please run clipppy locally to fix all the errors

**Comment by benoitmeriaux** [comment]
rustfmt is now failing because of changes done by clippy 🤔

**Review by zaidoon1 — APPROVED** [review]
LGTM

**Comment by benoitmeriaux** [comment]
@stanislav-tkach could you take a look, please ?

**Review by aleksuss — APPROVED** [review]

**Pull request merged**